### PR TITLE
SEQNG-676: Adding GHOST configuration.

### PR DESCRIPTION
**Pull request by sraaphorst**
Here are some changes I made to try to incorporate the GHOST configuration.

I'm not sure if this only comprises SEQNG-676 or another task as well.

### CODE REVIEW

File: modules/seqexec/server/src/main/scala/seqexec/server/ghost/GHOSTHeader.scala [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import gem.Observation
+import cats.effect.IO
+import seqexec.model.dhs.ImageFileId
+import seqexec.server.{InstrumentSystem, SeqAction}
+import seqexec.server.keywords._
+import seqexec.server.tcs.TcsKeywordsReader
+
+object GHOSTHeader {
+
+  def header(inst: InstrumentSystem[IO],
```
Comment (cquiroz) [inline modules/seqexec/server/src/main/scala/seqexec/server/ghost/GHOSTHeader.scala]: We don't need all the params if the `Header` is empty

File: modules/seqexec/server/src/main/scala/seqexec/server/SeqexecEngine.scala [hunk 2]
```diff
@@ -457,7 +463,9 @@ object SeqexecEngine extends SeqexecConfiguration {
                             failAt: Int,
                             odbQueuePollingInterval: Duration,
                             gpiGiapi: Giapi[IO],
-                            gpiGDS: Uri)
+                            ghostGiapi: Giapi[IO],
+                            gpiGDS: Uri,
+                            ghostGDS: Uri)
   def apply(httpClient: Client[IO], settings: Settings, c: SeqexecMetrics): SeqexecEngine = new SeqexecEngine(httpClient, settings, c)
 
   // Couldn't find this on Scalaz
```
Comment (cquiroz) [inline modules/seqexec/server/src/main/scala/seqexec/server/SeqexecEngine.scala]: this comment is likely stale

File: modules/seqexec/server/src/main/scala/seqexec/server/ghost/GHOSTHeader.scala (hunk 1 above)
Comment (sraaphorst) [reply modules/seqexec/server/src/main/scala/seqexec/server/ghost/GHOSTHeader.scala]: Can we get rid of all of them?

**Comment by cquiroz** [comment]
i think we are missing the part where `Settings` is populated
see https://github.com/gemini-hlsw/ocs3/blob/develop/modules/seqexec/server/src/main/scala/seqexec/server/SeqexecEngine.scala#L693

Also we need sensible defaults set in `app.conf`

File: modules/seqexec/server/src/main/scala/seqexec/server/SeqexecEngine.scala (hunk 2 above)
Comment (sraaphorst) [reply modules/seqexec/server/src/main/scala/seqexec/server/SeqexecEngine.scala]: I'll delete that :-)

File: modules/seqexec/server/src/main/scala/seqexec/server/ghost/GHOSTHeader.scala (hunk 1 above)
Comment (sraaphorst) [reply modules/seqexec/server/src/main/scala/seqexec/server/ghost/GHOSTHeader.scala]: I'm not sure which ones are safe to get rid of at this point... we have TCS keywords, I think, and I'm not sure about ObsKeywords. Do we need a GDSClient?

**Comment by jluhrs** [comment]
I think the changes ended up covering SEQNG-680

File: modules/seqexec/server/src/main/scala/seqexec/server/ghost/GHOSTHeader.scala (hunk 1 above)
Comment (cquiroz) [reply modules/seqexec/server/src/main/scala/seqexec/server/ghost/GHOSTHeader.scala]: We are not using any of them now. i'm surprsied WartRemover doesn't complaint. i'd remove them all and put them back when needed

Comment (sraaphorst) [reply modules/seqexec/server/src/main/scala/seqexec/server/ghost/GHOSTHeader.scala]: Sounds good to me!

**Comment by sraaphorst** [comment]
@jluhrs That's what I was thinking, too... I can rename this as 680, make another PR, and then do 676.

**Comment by sraaphorst** [comment]
Yes, you're right!

**Pull request closed without merging**